Use FetchRequestDestination enum for CORPViolationReportBody.destination

cdumez opened the pull request:
#### 6691152317a73aeaff9854f9f5457d8cfeb506fd
<pre>
Use FetchRequestDestination enum for CORPViolationReportBody.destination
<a href="https://bugs.webkit.org/show_bug.cgi?id=245185">https://bugs.webkit.org/show_bug.cgi?id=245185</a>

Reviewed by Brent Fulgham.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/fetch/FetchRequest.h:
* Source/WebCore/Modules/fetch/FetchRequest.idl:
* Source/WebCore/Modules/fetch/FetchRequestDestination.h: Copied from Source/WebCore/loader/CORPViolationReportBody.idl.
* Source/WebCore/Modules/fetch/FetchRequestDestination.idl: Copied from Source/WebCore/Modules/fetch/FetchRequest.idl.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/loader/CORPViolationReportBody.cpp:
(WebCore::CORPViolationReportBody::destination const): Deleted.
* Source/WebCore/loader/CORPViolationReportBody.h:
(WebCore::CORPViolationReportBody::destination const):
* Source/WebCore/loader/CORPViolationReportBody.idl:
* Source/WebCore/loader/CrossOriginEmbedderPolicy.cpp:

Canonical link: <a href="https://commits.webkit.org/254497@main">https://commits.webkit.org/254497@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3d64d6682e140946cd60ad61278ee2fd892eca9f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/89210 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33771 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/20025 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/98524 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/154838 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/93218 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32277 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/27821 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81569 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92993 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/94857 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25636 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/76129 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25566 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80501 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/80474 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/68540 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/30054 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/14536 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/29779 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15499 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3153 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33226 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/38457 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31929 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34590 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/DerivedSources.make:
```diff
@@ -313,7 +314,7 @@ JS_BINDING_IDLS := \
     $(WebCore)/Modules/filesystemaccess/FileSystemFileHandle.idl \
     $(WebCore)/Modules/filesystemaccess/FileSystemHandle.idl \
-	$(WebCore)/Modules/filesystemaccess/FileSystemSyncAccessHandle.idl \
+    $(WebCore)/Modules/filesystemaccess/FileSystemSyncAccessHandle.idl \
```
Someone must be using 8-space tabs!

brentfulgham commented on Source/WebCore/loader/CORPViolationReportBody.h:
```diff
@@ -40,7 +40,7 @@ class CORPViolationReportBody : public ReportBody {
     const AtomString& type() const final;
     const String& blockedURL() const { return m_blockedURL.string(); }
-    String destination() const;
+    FetchOptions::Destination destination() const { return m_destination; }
```
Nice.

brentfulgham reviewed (approved):
LGTM.

cdumez committed:
Use FetchRequestDestination enum for CORPViolationReportBody.destination
https://bugs.webkit.org/show_bug.cgi?id=245185

Reviewed by Brent Fulgham.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/fetch/FetchRequest.h:
* Source/WebCore/Modules/fetch/FetchRequest.idl:
* Source/WebCore/Modules/fetch/FetchRequestDestination.h: Copied from Source/WebCore/loader/CORPViolationReportBody.idl.
* Source/WebCore/Modules/fetch/FetchRequestDestination.idl: Copied from Source/WebCore/Modules/fetch/FetchRequest.idl.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/loader/CORPViolationReportBody.cpp:
(WebCore::CORPViolationReportBody::destination const): Deleted.
* Source/WebCore/loader/CORPViolationReportBody.h:
(WebCore::CORPViolationReportBody::destination const):
* Source/WebCore/loader/CORPViolationReportBody.idl:
* Source/WebCore/loader/CrossOriginEmbedderPolicy.cpp:

Canonical link: https://commits.webkit.org/254497@main
